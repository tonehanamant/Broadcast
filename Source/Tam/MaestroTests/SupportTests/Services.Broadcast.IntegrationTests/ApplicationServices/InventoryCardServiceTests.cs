using ApprovalTests;
using ApprovalTests.Reporters;
using IntegrationTests.Common;
using NUnit.Framework;
using Services.Broadcast.ApplicationServices;
using Services.Broadcast.Entities;
using System;

namespace Services.Broadcast.IntegrationTests.ApplicationServices
{
    [TestFixture]
    public class InventoryCardServiceTests
    {
        private readonly IInventoryCardService _InventoryCardService = IntegrationTestApplicationServiceFactory.GetApplicationService<IInventoryCardService>();

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetInitialData()
        {
            var inventoryCards = _InventoryCardService.GetInitialData(new DateTime(2019, 04, 01));

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(inventoryCards));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetInventoryCardsInventorySourceTest()
        {
            var inventoryCards = _InventoryCardService.GetInventoryCards(new InventoryCardFilterDto
            {
                InventorySourceId = 7
            }, new DateTime(2019, 2, 1));

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(inventoryCards));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetInventoryCardsInventorySourceNoDataTest()
        {
            var inventoryCards = _InventoryCardService.GetInventoryCards(new InventoryCardFilterDto
            {
                InventorySourceId = 8
            }, new DateTime(2019, 2, 1));

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(inventoryCards));
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void GetInventoryCardsQuarterTest()
        {
            var inventoryCards = _InventoryCardService.GetInventoryCards(new InventoryCardFilterDto
            {
                Quarter = new InventoryCardQuarter
                {
                    Quarter = 1,
                    Year = 2019
                }
            }, new DateTime(2019, 2, 1));

            Approvals.Verify(IntegrationTestHelper.ConvertToJson(inventoryCards));
        }
    }
}
