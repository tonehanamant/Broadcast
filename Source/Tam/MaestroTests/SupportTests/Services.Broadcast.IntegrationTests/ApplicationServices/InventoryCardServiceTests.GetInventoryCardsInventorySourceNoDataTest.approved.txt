[
  {
    "HasRatesAvailableForQuarter": false,
    "HouseholdImpressions": null,
    "InventoryPostingBooks": [],
    "InventorySourceId": 8,
    "InventorySourceName": "MLB",
    "IsUpdating": false,
    "LastUpdatedDate": null,
    "Quarter": {
      "Quarter": 1,
      "Year": 2019
    },
    "RatesAvailableFromQuarter": null,
    "RatesAvailableToQuarter": null,
    "TotalDaypartCodes": 0,
    "TotalMarkets": 0,
    "TotalStations": 0,
    "TotalUnits": 0
  }
]